heksher/setting.py

```diff
@@ -47,8 +46,19 @@ def __init__(self, name: str, type, configurable_features: Sequence[str],
 
         self.last_ruleset: Optional[RuleSet] = None
 
+        self._callbacks: List[Callable[[Setting, Sequence[str], T], T]] = []
+
         heksher.main_client.Main.add_settings((self,))
 
+    def add_callback(self, callback: Callable[[Setting, Sequence[str], T], T]) -> None:
```

Comment:
1. I think `validator` is a more fitting name
a. it's more consistent with pydantic and other libraries
b. "callbacks" usually have side effects, if these functions have side effects there'll be chaos
2. since we usually need the value more than the contexts or the setting, I think a better signature would be `value, rule, setting`, then we could do:
```
@setting.add_callback
def foo(value: str, *_):
  return value.lower()
```

Reply:
also, why is the callback accepting the rule as a sequence of strings? shouldn't it be a dict?

Reply:
I suggest that this function return `callback`, so that it can be used as a decorator.